sno/gpkg.py

```diff
@@ -185,14 +186,11 @@ def geom_cols(db, table):
     return tuple(r[0] for r in q)
 
 
-def geom_to_ogr(gpkg_geom, parse_srs=False):
+def _validate_geom(gpkg_geom):
```

Comment:
once again I've just realised this PR duplicates some effort from the postgis branch :( any ETA on finishing that off?

I've added tests in 0a14e9c058c083d31daf6baeb4d56dc1f64a7508 comprehensively covering envelope and endianness varieties. it doesn't cover SRID since I haven't implemented any handling of that - it produces WKB (not EWKB) and thus strips the SRID.

I haven't covered any exotic curve geometries etc because they don't really seem like edge cases for this, since we don't bother probing inside the WKB itself.

Reply:
> any ETA on finishing that off?

After internal data format changes. It got real hard without them.